.github/workflows/ci.yaml

```diff
@@ -0,0 +1,42 @@
+name: CI
+on:
+  pull_request:
+    paths-ignore:
+      - 'docs/**'
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - 'docs/**'
+
+jobs:
+  unittest:
+    name: unit tests
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python 
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.11'
+          cache: pip
+
+      - name: Get library from main
+        run: |
+              git clone https://github.com/smart-on-fhir/cumulus-library.git
+              cd cumulus-library
+              pip install -e .
+              cd ..
```

Comment:
nit: since you don't actually end up using the checkout, you could get away with this:
```
      - name: Get library from main
        run: pip install git+https://github.com/smart-on-fhir/cumulus-library.git
```

Do you maybe want this after the "install dependencies" stanza? 1) you get the latest pip first that way and 2) what if your pyproject has a dependency pin like "cumulus-library >=2, <3" -- it could silently undo the install-from-main here. (I know this PR also unpins the library a bit, but still - it's the silent nature of it here that could hide issues with the ordering)

Reply:
oooh fancy, i like it.

I don't want this in the dependencies stanza, since this is currently pinned to 3.0, which is not yet released.

Reply:
when it :is: released i'm going to remove this stanza entirely.